### DISCUSSION TITLE
Boolean value as value

### DISCUSSION THREAD

**Discussion by Jack-Works** [Jack-Works/proposal-enum#7] [General]
In flow.js and [rbuckton's proposal](https://github.com/rbuckton/proposal-enum#examples), boolean value is supported as the enum key.

Boolean only have 2 value in the value space so I'm wondering if it is really useful, but it also has no harm if we support it.

```ts
enum Toggle {
    on = true, off = false
}
```

Should we support this use case?

**Comment by dnalborczyk** [comment]
> if it is really useful, but it also has no harm if we support it.

some might find it useful. constraining the syntax to disallow booleans is probably unnecessary overhead.

**Comment by ljharb** [comment]
Can an enum only have one type of thing? What's wrong with an enum that has values of true, false, 0, 1, and `'string'`, for example?

**Reply by Jack-Works** [reply]
> Can an enum only have one type of thing? What's wrong with an enum that has values of true, false, 0, 1, and `'string'`, for example?

That's probably OK but I doubt if it is a foot gun. I have a topic for that at #8

**Comment by Jack-Works** [comment]
I'm allowing boolean.
